core/src/test/scala/unit/kafka/server/ListOffsetsRequestTest.scala

```diff
@@ -162,11 +162,15 @@ class ListOffsetsRequestTest extends BaseRequestTest {
     val partitionToLeader = TestUtils.createTopic(zkClient, topic, numPartitions = 1, replicationFactor = 3, servers)
     val firstLeaderId = partitionToLeader(partition.partition)
 
-    TestUtils.generateAndProduceMessages(servers, topic, 10)
+    // produce in 2 batches to ensure the max timestamp matches the last message
+    TestUtils.generateAndProduceMessages(servers, topic, 9)
+    Thread.sleep(10)
+    TestUtils.generateAndProduceMessages(servers, topic, 1)
```

Comment:
It this really necessary? If yes, could we remove the `sleep`?

Reply:
This test had become flaky because multiple messages could have the same timestamp. This pause ensured the last message had the highest timestamp. I've removed the sleep now and explicitly set the timestamp in the test messages.